lib/api/src/tests/stories.test.js

```diff
@@ -405,95 +410,33 @@ describe('stories API', () => {
         isComponent: true,
       });
     });
+  });
 
-    it('navigates to the first story in the store if there is none selected', async () => {
-      const navigate = jest.fn();
-      const store = createMockStore({ viewMode: 'story' });
-
-      const {
-        api: { setStories },
-      } = initStories({ store, navigate, provider });
-
-      await setStories(storiesHash);
-      expect(navigate).toHaveBeenCalledWith('/story/a--1');
-    });
-
-    it('navigates to the first story in the store if a non-existent story was selected', async () => {
-      const navigate = jest.fn();
-      const store = createMockStore({ viewMode: 'story', storyId: 'random' });
-
-      const {
-        api: { setStories },
-      } = initStories({ store, navigate, provider });
-
-      await setStories(storiesHash);
-      expect(navigate).toHaveBeenCalledWith('/story/a--1');
-    });
-
-    it('navigates to the first leaf story if a story exists but it is not a leaf story(1)', async () => {
-      const navigate = jest.fn();
-      const store = createMockStore({ viewMode: 'story', storyId: 'b' });
-
-      const {
-        api: { setStories },
-      } = initStories({ store, navigate, provider });
-
-      await setStories(storiesHash);
-      expect(navigate).toHaveBeenCalledWith('/story/b-c--1');
-    });
-
-    it('navigates to the first leaf story if a story exists but it is not a leaf story(2)', async () => {
-      const navigate = jest.fn();
-      const store = createMockStore({ viewMode: 'story', storyId: 'b-d' });
-
-      const {
-        api: { setStories },
-      } = initStories({ store, navigate, provider });
-
-      await setStories(storiesHash);
-      expect(navigate).toHaveBeenCalledWith('/story/b-d--1');
-    });
-
-    it('does not navigate if a existing story was selected', async () => {
-      const navigate = jest.fn();
-      const store = createMockStore({ viewMode: 'story', storyId: 'b-c--1' });
-
-      const {
-        api: { setStories },
-      } = initStories({ store, navigate, provider });
-
-      await setStories(storiesHash);
-      expect(navigate).not.toHaveBeenCalled();
-    });
-
-    it('navigates to the settings page if a existing page was selected', async () => {
-      const navigate = jest.fn();
-      const store = createMockStore({ viewMode: 'settings', storyId: 'about' });
+  // Can't currently run these tests as cannot set this on the events
+  // describe('CURRENT_STORY_WAS_SET event', () => {
+  //   it('navigates to the story', async () => {
+  //     const navigate = jest.fn();
+  //     const api = EventEmitter();
+  //     const store = createMockStore({});
+  //     const { init } = initStories({ store, navigate, provider, fullAPI: api });
 
-      const {
-        api: { setStories },
-      } = initStories({ store, navigate, provider });
+  //     init();
+  //     api.emit(CURRENT_STORY_WAS_SET, { storyId: 'a--1', viewMode: 'story' });
```

Comment:
@ndelangen -- how can I emit an event here that is treated as coming from the local ref? Do you have a pattern for this in tests?

Reply:
The `this` ref is added here:
https://github.com/storybookjs/storybook/blob/8a20d1c178cd7aeb1acb41bd07accfde9ddb54bc/lib/channel-postmessage/src/index.ts#L43-L44

Testing this would require a mod on the EventEmitter..

We could do this:
https://github.com/primus/EventEmitter3#contextual-emits

Reply:
I found a decent solution

Reply:
LGTM